app/javascript/packs/containers/BuildingContainer.jsx

```diff
@@ -1,34 +1,56 @@
 import React from 'react';
-import Question from '../components/Question';
-import { getQuestionsByBuilding } from '../selectors/questionsSelector';
-import { bindActionCreators } from 'redux';
+import AnswerModeContainer from './AnswerModeContainer';
+import DelegateModeContainer from './DelegateModeContainer';
+import ReviewModeContainer from "./ReviewModeContainer";
+import QuestionnaireControls from "../components/QuestionnaireControls";
+
 import { connect } from 'react-redux';
 
+import { Route, Switch } from 'react-router-dom';
+
 class BuildingContainer extends React.Component {
   render() {
+    const currentPath = this.props.match.url;
+    const propsToPass = {
+      building: this.props.building
+    };
+
     return (
       <div>
         <p>Building container!!</p>
         <p>ID: {this.props.match.params.bId}</p>
         <p>Name: {this.props.building.name}</p>
+
+        <QuestionnaireControls currentPath={currentPath} />
+
         <hr />
-        {this.props.questions.map((question) => {
-          // Only display non-dependent questions initially
-          if (!question.parent_option_id) {
-            return (<Question key={question.id}
-                              building_id={this.props.building.id}
-                              {...question} />);
-          }
-        })}
+
+        <Switch>
+          <Route path={`${currentPath}/edit`}
+                 render={renderWithProps(AnswerModeContainer, propsToPass)} />
+          <Route path={`${currentPath}/delegate`}
+                 render={renderWithProps(DelegateModeContainer, propsToPass)} />
+          <Route path={`${currentPath}/review`}
+                 render={renderWithProps(ReviewModeContainer, propsToPass)} />
+        </Switch>
       </div>
     );
   }
 }
 
+/**
+ * Helper method - returns a render function that renders the component with the given props passed down to it.
+ * Used by Routes.
+ */
+function renderWithProps(ComponentName, props) {
+  return () => (
+    <ComponentName {...props} />
+  );
+}
+
 function mapStateToProps(state, ownProps) {
   return {
-    building: state.buildings[ownProps.match.params.bId],
-    questions: getQuestionsByBuilding(ownProps.match.params.bId, state)
+    building: state.buildings[ownProps.match.params.bId]
```

Comment:
This is something I'm not sure about -- since the BuildingContainer has child containers for edit/delegate/review modes now, I'm having the child containers get the questions (and possibly other questionnaire-related info) themselves in `mapStateToProps`, so that the BuildingContainer is only responsible for getting `building` from store. 

The problem is that each of the child containers could potentially end up having the same code in mapStateToProps (not sure if this is good practice?). However, the alternative would be to have BuildingContainer get all the questionnaire-related stuff in its `mapStateToProps` and pass them down to the child containers, which would no longer really have to read from store at all. This seems a bit weird too... 😕

Reply:
I haven't done a full review of the container at this current state, but I like [Dan's mini-rant](https://twitter.com/dan_abramov/status/944931372820549632) about these sort of "_is this best practice in Redux/React/*?_" questions. It's totally okay to have "containers" nested within containers (we do that anyway between `BuildingContainer` and `OptionsContainer`). I think the way you've designed this, for now, is fine unless in the future we see so many redundancies across `BuildingContainer` and one of its child containers that it would be more practical to pass that into props or to consider lifting store up and out the child.

Reply:
leaving my 2 cents here but I personally am a fan of mapping state to props in react components unless it's a teeny tiny visual component - then i don't care. It may feel redundant but it also makes it a lot more flexible to changes; no stupid propagation of props for days down a tunnel of containers.